modules/infra/control/loopback.c

```diff
@@ -146,6 +152,15 @@ static void iface_loopback_poll(evutil_socket_t, short reason, void *ev_iface) {
 	rte_pktmbuf_trim(mbuf, read_len - len);
 	eth = rte_pktmbuf_mtod(mbuf, struct rte_ether_hdr *);
 
+	if (!rte_is_same_ether_addr(&eth->src_addr, &lo->mac)) {
+		LOG(NOTICE,
+		    "src_addr(" ETH_F ") differs from interface address(" ETH_F "). updating %s",
+		    &eth->src_addr,
+		    &lo->mac,
+		    iface->name);
+		loopback_mac_set(iface, &eth->src_addr);
```

Comment:
I am not comfortable with this. If really needed we should subscribe to netlink notifications for that network device and update our internal mac address accordingly.

Also, you should *NEVER* call any log function in the packet reception path. In this case, we are still in the control path so that is *acceptable*.

In any case, I think the correct fix is to update the mac address upon notification from the kernel *or alternatively*, disable systemd-networkd for these interfaces.